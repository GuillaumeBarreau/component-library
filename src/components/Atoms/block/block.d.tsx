export type TVariants = "div" | "section" | "main" | undefined;

export interface ComponentBlockProps {
  children?: React.ReactNode;
  className?: string;
  variant?: TVariants;
}
