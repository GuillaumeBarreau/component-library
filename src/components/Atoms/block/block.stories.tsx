import { Block } from "./block";
import { Meta } from "@storybook/react/types-6-0";
import { ComponentBlockProps } from "./block.d";

export default {
  title: "block",
  component: Block,
  parameters: {
    a11y: {
      // optional selector which element to inspect
      element: "#root",
      // optional flag to prevent the automatic check
      manual: false,
    },
  },
} as Meta;

const Template: any = (args: JSX.IntrinsicAttributes & ComponentBlockProps) => (
  <>
    <section style={{ display: "flex", justifyContent: "center" }}>
      <Block {...args} />
    </section>
  </>
);

export const BlockDefault = Template.bind({});
BlockDefault.args = {
  children: "block",
};
