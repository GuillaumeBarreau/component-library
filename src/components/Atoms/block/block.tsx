import React from "react";
import classNames from "classnames";
import { ComponentBlockProps } from "./block.d";
import "./block.css";

export const Block: React.FC<ComponentBlockProps> = (props) => {
  const { children, className, variant = "div" } = props;
  const prefixCls: string = "block";

  const classes = classNames(prefixCls, {}, className);

  const blockNode = React.createElement(
    variant,
    {
      className: classes,
    },
    children
  );

  return blockNode;
};
