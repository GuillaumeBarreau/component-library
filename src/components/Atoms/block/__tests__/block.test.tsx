import React from "react";
import { shallow } from "enzyme";
import { Block } from "../block";
import { ComponentBlockProps } from "../block.d";

describe("Block", () => {
  let defaultProps: JSX.IntrinsicAttributes & ComponentBlockProps;

  beforeEach(() => {
    defaultProps = {
      children: "Block",
    };
  });

  it("renders correctly", () => {
    const wrapper = shallow(<Block {...defaultProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("mount correctly", () => {
    const wrapper = shallow(<Block {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
  });

  it("Should contain defaultProps:children correctly", () => {
    const wrapper = shallow(<Block {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain("Block");
  });
});
