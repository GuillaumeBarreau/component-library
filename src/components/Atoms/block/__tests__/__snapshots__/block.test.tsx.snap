// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders correctly 1`] = `
"<Component className=\\"block\\">
  Block
</Component>"
`;
